modules/runner/src/main/ts/reporter/MochaReporter.ts

```diff
@@ -0,0 +1,124 @@
+import { LoggedError } from '@ephox/bedrock-common';
+import Promise from '@ephox/wrap-promise-polyfill';
+import { Context, MochaOptions, Runner as MochaRunner, Test } from 'mocha';
+import { Lock } from '../core/Lock';
+import { noop } from '../core/Utils';
+import { Reporter, TestReporter } from './Reporter';
+
+type LoggedError = LoggedError.LoggedError;
+
+export interface MochaReporterOptions {
+  readonly reporter: Reporter;
+  readonly numTests: number;
+  readonly onPass: () => void;
+  readonly onSkip: () => void;
+  readonly onFailure: (err: LoggedError) => void;
+}
+
+const defaultOptions: Partial<MochaReporterOptions> = {
+  onPass: noop,
+  onSkip: noop,
+  onFailure: noop
+};
+
+const isTestEqual = (test1: Test | undefined, test2: Test | undefined) => {
+  if (test1 === test2) {
+    return true;
+  } else if (test1 === undefined || test2 === undefined) {
+    return false;
+  } else {
+    return test1.parent === test2.parent
+      && test1.ctx === test2.ctx
+      && test1.file === test2.file
+      && test1.title === test2.title;
+  }
+};
+
+export class BedrockMochaReporter extends Mocha.reporters.Base {
+  constructor(runner: MochaRunner, options: MochaOptions) {
+    super(runner, options);
+
+    const opts: MochaReporterOptions = {
+      ...defaultOptions,
+      ...options.reporterOptions
+    };
+    const reporter = opts.reporter;
+
+    const lock = Lock();
+    let currentTest: Test | undefined;
+    let lastError: LoggedError | undefined;
+    let report: TestReporter;
+    let aborted = false;
+
+    const processResult = (test?: Test): Promise<void> => {
+      if (!test || test.isFailed()) {
+        // We're guaranteed to have an error if the state is a failure
+        const error = lastError as LoggedError;
+        return report.fail(error).then(() => opts.onFailure(error));
+      } else if (test.isPending()) {
+        return report.skip(test.title).then(opts.onSkip);
+      } else if (test.isPassed()) {
+        return report.pass().then(opts.onPass);
+      } else {
+        return Promise.resolve();
+      }
+    };
+
+    // Setup root hooks, as for the reporter we need the start/pass/fail to be async meaning we can't use events
+    mocha.rootHooks({
+      beforeEach() {
+        return lock.execute(() => report.start());
+      },
+      afterEach(this: Context) {
+        const test = this.currentTest;
+        return lock.execute(() => processResult(test));
+      }
+    });
+
+    const onTestStart = (test: Test) => {
+      if (!isTestEqual(test, currentTest)) {
+        currentTest = test;
+        const testTitle = test.titlePath().slice(0, -1).join(' / ') + ' ' + test.title;
```

Comment:
It's not a string for starters, it's an array. So this removes the last entry from the array and then joins it back together so that we end up with something like this:
```
Parent / Current - Test
```

Reply:
Oh yep, I misread this - all good